SC1008 Shell detection fails on "#!/usr/bin/env -S bash"

#### For bugs
- Rule Id: SC1008
- My shellcheck version: "online" (2021-02-10, master at 15ff87cf80bfcc23fa4a3d0412594e17959dd1b0)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related


#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env -S bash
```

#### Here's what shellcheck currently says:
```
Line 1:
#!/usr/bin/env -S bash
^-- SC1008: This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
```

#### Here's what I wanted or expected to see:
No output, `bash` is deduced from the shebang just like with `#!/usr/bin/env bash`.

--- a/src/ShellCheck/Parser.hs
+++ b/src/ShellCheck/Parser.hs
@@ -3190,6 +3190,7 @@ prop_readScript2 = isWarning readScript "#!/bin/bash\r\necho hello world\n"
 prop_readScript3 = isWarning readScript "#!/bin/bash\necho hello\xA0world"
 prop_readScript4 = isWarning readScript "#!/usr/bin/perl\nfoo=("
 prop_readScript5 = isOk readScript "#!/bin/bash\n#This is an empty script\n\n"
+prop_readScript6 = isOk readScript "#!/usr/bin/env -S X=FOO bash\n#This is an empty script\n\n"
 readScriptFile sourced = do
     start <- startSpan
     pos <- getPosition
@@ -3231,13 +3232,14 @@ readScriptFile sourced = do
 
   where
     basename s = reverse . takeWhile (/= '/') . reverse $ s
+    skipFlags = dropWhile ("-" `isPrefixOf`)
     getShell sb =
         case words sb of
             [] -> ""
             [x] -> basename x
-            (first:second:_) ->
+            (first:args) ->
                 if basename first == "env"
-                    then second
+                    then fromMaybe "" $ find (notElem '=') $ skipFlags args
                     else basename first
 
     verifyShebang pos s = do

